new/detector/composition/ruby/ruby.go

```diff
@@ -192,10 +192,14 @@ func (composition *Composition) DetectFromFileWithTypes(file *file.FileInfo, det
 	var result []*detectortypes.Detection
 	for _, detectorType := range detectorTypes {
 		rule := composition.rules[detectorType]
+		sanitizerRuleID := ""
+		if rule != nil {
+			sanitizerRuleID = rule.SanitizerRuleID
+		}
 		detections, err := evaluator.Evaluate(
 			tree.RootNode(),
 			detectorType,
-			rule.SanitizerRuleID,
+			sanitizerRuleID,
 			settings.NESTED_SCOPE,
```

Comment:
I'll re-visit this tomorrow - @didroe I'm guessing a `nil` rule here is not a real-world possibility?

Reply:
You should always have a rule as the `detectorTypes` were gathered from the rules in the first place